src/catch2/internal/catch_compiler_detections.hpp

```diff
@@ -0,0 +1,26 @@
+
+//              Copyright Catch2 Authors
+// Distributed under the Boost Software License, Version 1.0.
+//   (See accompanying file LICENSE_1_0.txt or copy at
+//        https://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef CATCH_COMPILER_DETECTIONS_HPP_INCLUDED
+#define CATCH_COMPILER_DETECTIONS_HPP_INCLUDED
+
+#if defined(__GNUC__) && !defined(__clang__) && !defined(__ICC) && !defined(__CUDACC__) && !defined(__LCC__)
```

Comment:
This approach isn't going to scale well, as pretty much every new or obscure compiler masquerades for GCC, so this line is going to keep getting longer and longer.

Instead, let's have a macro `CATCH_COMPILER_DETECTED`, that get's set when a compiler was picked. This would simplify GCC (and also clang) to something like 

```cpp
#if defined(__GNUC__) and !defined(CATCH_COMPILER_DETECTED)
#define CATCH_COMPILER_GCC
#endif
```